Remove outdated information for transactional.id (#2524)
spring-projects/spring-kafka: spring-kafka-docs/src/main/asciidoc/kafka.adoc

Transactions are enabled by providing the `DefaultKafkaProducerFactory` with a `transactionIdPrefix`.
In that case, instead of managing a single shared `Producer`, the factory maintains a cache of transactional producers.
When the user calls `close()` on a producer, it is returned to the cache for reuse instead of actually being closed.

> The `transactional.id` property of each producer is `transactionIdPrefix` + `n`, where `n` starts with `0` and is incremented for each new producer.

Author: ```suggestion
The `transactional.id` property of each producer is `transactionIdPrefix` + `n`, where `n` starts with `0` and is incremented for each new producer.
In previous versions of Spring Kafka, the `transactional.id` was generated differently for transaction started by a listener container with a record-based listener to support fencing zombies, which is not necessary anymore with EOSMode V2 being the only option starting with 3.0.0.
For applications running with multiple instances, the `transactionIdPrefix` should be unique per instance.
```

Transactions are enabled by providing the `DefaultKafkaProducerFactory` with a `transactionIdPrefix`.
In that case, instead of managing a single shared `Producer`, the factory maintains a cache of transactional producers.
When the user calls `close()` on a producer, it is returned to the cache for reuse instead of actually being closed.
The `transactional.id` property of each producer is `transactionIdPrefix` + `n`, where `n` starts with `0` and is incremented for each new producer.

> In previous versions of Spring Kafka, the `transactional.id` was generated differently for transaction started by a listener container with a record-based listener to support fencing zombies, which is not necessary anymore with `EOSMode.V2` being the only option starting with 3.0.0.
> For applications running with multiple instances, the `transactionIdPrefix` should be unique per instance.

Reviewer: ```suggestion
In previous versions of Spring for Apache Kafka, the `transactional.id` was generated differently for transactions started by a listener container with a record-based listener, to support fencing zombies, which is not necessary any more, with `EOSMode.V2` being the only option starting with 3.0.
For applications running with multiple instances, the `transactionIdPrefix` must be unique per instance.
```
